src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -100,4 +100,33 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
             return response;
         }
     }
+    
+    // Custom web hooks for extensions. 
+    public class HookController : ApiController
+    {
+        private readonly WebScriptHostManager _scriptHostManager;
+
+        public HookController(WebScriptHostManager scriptHostManager)
+        {
+            _scriptHostManager = scriptHostManager;
+        }
+                
+        [Route("hook/{name}")]
```

Comment:
I think this route should be under our reserved admin route, so we don't conflict with any user routes. E.g. `admin/hooks/{name}`